src/site/content/en/blog/nfc/index.md

```diff
@@ -0,0 +1,911 @@
+locked. For non visible web pages, receiving and pushing NFC content are
+suspended, and resumed when a web page becomes visible again.
+
+Thanks to the [Page Visibility API], it is possible to track when document
+visibility changes.
+
+```js
+document.onvisibilitychange = event => {
+  if (document.hidden) {
+    // All NFC operations are automatically suspended when document is hidden.
+  } else {
+    // All NFC operations are resumed, if needed.
+  }
+};
+```
+
+## Cookbook {: #cookbook }
+
+Because it is always better to start coding with examples, here are some that
+may help.
+
+### Check for permission
+
+The [Permissions API] allows checking whether the `"nfc"` permission was granted
+previously. This example shows how to scan NFC tags without user interaction if
+access was previously granted, or show a button otherwise. Note that the same
+mechanism works for writing NFC tags as it uses the same permission under the
+hood.
+
+```js
+const reader = new NDEFReader();
+
+async function startScanning() {
+  await reader.scan();
+  reader.onreading = event => {
+    /* handle NDEF messages */
+  };
+}
+
+const nfcPermissionStatus = await navigator.permissions.query({ name: "nfc" });
+if (permissionStatus.state === "granted") {
+  // NFC access was previously granted, so we can start NFC scanning now.
+  startScanning();
+} else {
+  // Show a "scan" button.
+  document.querySelector("#scanButton").style.display = "block";
+  document.querySelector("#scanButton").onclick = event => {
+    // Prompt user to allow UA to send and receive info when they tap NFC devices.
+    startScanning();
+  };
+}
+```
+
+### Abort NFC operations
+
+Using the [AbortController] primitive makes it easy to abort NFC operations and
+other operations that use it. The example below shows you how to pass the
+`signal` of an AbortController through the options of NDEFReader `scan()` and
+NDEFWriter `write()` methods and abort both NFC operations at the same time.
+
+```js
+const abortController = new AbortController();
+abortController.signal.onabort = event => {
+  // All NFC operations have been aborted.
+};
+
+const reader = new NDEFReader();
+await reader.scan({ signal: abortController.signal });
+
+const writer = new NDEFWriter();
+await writer.push("Hello world", { signal: abortController.signal });
+
+document.querySelector("#abortButton").onclick = event => {
+  abortController.abort();
+};
+```
+
+### Read and write a text record
+
+The text record `data` can be decoded with a `TextDecoder` instantiated with the
+record `encoding` property. Note that the language of the text record is
+available through its `lang` property.
+
+```js
+function readTextRecord(record) {
+  console.assert(record.recordType === "text");
+  const textDecoder = new TextDecoder(record.encoding);
+  console.log(`Text: ${textDecoder.decode(record.data)} (${record.lang})`);
+}
+```
+
+Writing a simple text record can be done by passing a string to the NDEFWriter
```

Comment:
'To write a simple text record, pass a string...' (Changes passive to active voice.)

Reply:
Done